src/manifolds/ProductManifold.jl

```diff
@@ -736,6 +736,23 @@ for TP in [ProductRepr, ArrayPartition]
                     ),
                 )
             end
+            function inverse_retract!(
```

Comment:
oh, did we forget to implement the mutating inverse retraction for product manifolds? Interesting!

Reply:
Yes, I was very surprised that we didn't have that.

Reply:
So am I, good that we have it now, then :)